app/src/components/misc/PersonaSurvey/ModalFooter.tsx

```diff
@@ -1,88 +1,84 @@
-import React from "react";
+import React, { useMemo } from "react";
 import { useSelector, useDispatch } from "react-redux";
-import { useNavigate } from "react-router-dom";
 import { getUserPersonaSurveyDetails, getAppMode } from "store/selectors";
 import { actions } from "store";
 import { Col, Row } from "antd";
 import { RQButton } from "lib/design-system/components";
-import { OptionsConfig, SurveyConfig } from "./config";
-import APP_CONSTANTS from "config/constants";
+import { SurveyConfig } from "./config";
+import { handleSurveyNavigation } from "./actions";
+import { submitAttrUtil } from "utils/AnalyticsUtils";
+import { QuestionnaireType, SurveyPage } from "./types";
 //@ts-ignore
 import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
-import { submitAttrUtil } from "utils/AnalyticsUtils";
+import PATHS from "config/constants/sub/paths";
+import APP_CONSTANTS from "config/constants";
 import { trackPersonaQ1Completed } from "modules/analytics/events/misc/personaSurvey";
 import "./index.css";
-import { QuestionnaireType } from "./types";
-import PATHS from "config/constants/sub/paths";
 
 interface FooterProps {
-  currentPage: number;
+  currentPage: SurveyPage;
   callback: () => void;
   isSurveyModal: boolean;
 }
 
 export const SurveyModalFooter: React.FC<FooterProps> = ({ currentPage, callback, isSurveyModal }) => {
   const dispatch = useDispatch();
-  const navigate = useNavigate();
   const userPersona = useSelector(getUserPersonaSurveyDetails);
   const appMode = useSelector(getAppMode);
   const currentQuestionnaire = SurveyConfig[currentPage]?.render;
+  const surveyLength = useMemo(() => Object.keys(SurveyConfig).length, []);
   const isSharedListUser = window.location.href.includes(PATHS.SHARED_LISTS.VIEWER.RELATIVE);
 
-  const disableContinue = () => {
-    if (currentPage === 0) return false;
-    if (userPersona[OptionsConfig[currentQuestionnaire as QuestionnaireType]?.key]?.length) return false;
+  const disableContinue = useMemo(() => {
+    if (currentPage === SurveyPage.GETTING_STARTED) return false;
+    if (userPersona[currentQuestionnaire as QuestionnaireType]?.length) return false;
     return true;
-  };
+  }, [currentPage, currentQuestionnaire, userPersona]);
 
   const handleMoveToNextPage = () => {
+    const index = Object.keys(SurveyConfig).indexOf(currentPage);
```

Comment:
how `index` is helping in navigation?

Reply:
index is used here to make decisions on last step of persona survey, wether to move to recommendation screen or not , or the survey is from workspace onboarding then perform its necessary actions on survey completion.